core/src/main/java/io/grpc/internal/DelayedClientTransport.java

```diff
@@ -155,7 +155,8 @@ public final ClientStream newStream(MethodDescriptor<?, ?> method, Metadata head
       }
       if (picker != null) {
         while (true) {
-          PickResult pickResult = picker.pickSubchannel(callOptions.getAffinity(), headers);
+          PickResult pickResult = picker.pickSubchannel(
+              new PickSubchannelArgsImpl(method, headers, callOptions));
```

Comment:
It's possible that it is and I've made some stupid mistake, the `DelayedClientTransportTest` and one other tests were failing. I'll look into this again later today.